test/matrix.cc

```diff
@@ -391,6 +393,150 @@ TEST(Matrix, partial_matrix) {
   EXPECT_EQ(found, 2) << " partial result did not find 2 results as expected";
 }
 
+// slim dowm matrix response: https://github.com/valhalla/valhalla/pull/3987
+const auto test_matrix_default = R"({
+    "sources":[
+      {"lat":52.103948,"lon":5.06813}
+    ],
+    "targets":[
+      {"lat":52.106126,"lon":5.101497},
+      {"lat":52.100469,"lon":5.087099},
+      {"lat":52.103105,"lon":5.081005},
+      {"lat":52.094273,"lon":5.075254}
+    ],
+    "costing":"auto"
+  })";
+
+TEST(Matrix, default_matrix) {
+  loki_worker_t loki_worker(config);
+
+  Api request;
+  ParseApi(test_matrix_default, Options::sources_to_targets, request);
+
+  loki_worker.matrix(request);
+
+  GraphReader reader(config.get_child("mjolnir"));
+
+  sif::mode_costing_t mode_costing;
+  mode_costing[0] =
+      CreateSimpleCost(request.options().costings().find(request.options().costing_type())->second);
+
+  TimeDistanceMatrix timedist_matrix;
+  std::vector<TimeDistance> results =
+      timedist_matrix.SourceToTarget(*request.mutable_options()->mutable_sources(),
+                                     *request.mutable_options()->mutable_targets(), reader,
+                                     mode_costing, sif::TravelMode::kDrive, 400000.0);
+
+  auto response = serializeMatrix(request, results, 1.0);
+
+  rapidjson::Document json;
+  json.Parse(response);
+
+  ASSERT_FALSE(json.HasParseError());
+
+  EXPECT_TRUE(json.HasMember("sources_to_targets"));
+
+  if (json.HasMember("sources_to_targets")) {
+
+    EXPECT_EQ(json["sources_to_targets"].GetArray()[0][0].MemberCount(), 4)
+        << " contains 4 keys i.e, distance, time, to_index and from_index";
```

Comment:
it's great that you put those here for explanation what it really does, but it looks like codecov thinks it's not executed (which it really probably even isn't). I'd convert those to comments, then codecov shuts up and coverage should be 100% again.

Reply:
Addressed!